crates/bgp-pkt/src/wire/tests/bgp_ls.rs

```diff
@@ -0,0 +1,398 @@
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x01, 0x00, 0x28, 0x01,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x1b, 0x02, 0x00, 0x00,
+        0x04, 0x00, 0x01, 0x00, 0x00, 0x02, 0x01, 0x00, 0x04, 0x00, 0x00, 0x00, 0x00, 0x02, 0x03,
+        0x00, 0x07, 0x00, 0x00, 0x00, 0x00, 0x00, 0x09, 0x01, 0x00, 0x01, 0x00, 0x28, 0x01, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x1b, 0x02, 0x00, 0x00, 0x04,
+        0x00, 0x01, 0x00, 0x00, 0x02, 0x01, 0x00, 0x04, 0x00, 0x00, 0x00, 0x00, 0x02, 0x03, 0x00,
+        0x07, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x03,
+    ];
+
+    let good = PathAttribute::from(
+        true,
+        false,
+        false,
+        true,
+        PathAttributeValue::MpReach(MpReach::BgpLs {
+            next_hop: IpAddr::V6(Ipv6Addr::from_str("2001:db8:1::1").unwrap()),
+            nlri: vec![
+                BgpLsNlri {
+                    path_id: None,
+                    value: BgpLsNlriValue::Node(BgpLsNlriNode {
+                        protocol_id: BgpLsProtocolId::IsIsLevel1,
+                        identifier: 0,
+                        local_node_descriptors: BgpLsNodeDescriptorTlv::Local(vec![
+                            BgpLsNodeDescriptorSubTlv::AutonomousSystem(65536),
+                            BgpLsNodeDescriptorSubTlv::BgpLsIdentifier(0),
+                            BgpLsNodeDescriptorSubTlv::IgpRouterId(vec![0, 0, 0, 0, 0, 9, 1]),
+                        ]),
+                    }),
+                },
+                BgpLsNlri {
+                    path_id: None,
+                    value: BgpLsNlriValue::Node(BgpLsNlriNode {
+                        protocol_id: BgpLsProtocolId::IsIsLevel1,
+                        identifier: 0,
+                        local_node_descriptors: BgpLsNodeDescriptorTlv::Local(vec![
+                            BgpLsNodeDescriptorSubTlv::AutonomousSystem(65536),
+                            BgpLsNodeDescriptorSubTlv::BgpLsIdentifier(0),
+                            BgpLsNodeDescriptorSubTlv::IgpRouterId(vec![0, 0, 0, 0, 0, 1, 3]),
+                        ]),
+                    }),
+                },
+            ],
+        }),
+    )
+    .unwrap();
+
+    test_parsed_completely_with_one_input(
+        &good_wire,
+        &mut BgpParsingContext::new(
+            true,
+            HashMap::new(),
+            HashMap::new(),
+            false,
+            false,
+            false,
+            false,
+        ),
+        &good,
+    );
+    test_write(&good, &good_wire)?;
+
+    Ok(())
+}
+
+#[test]
+pub fn test_bgp_ls_attr_parse() {
+    let value = BgpLsAttribute {
+        tlvs: vec![BgpLsAttributeTlv::LinkName("My Super Link".to_string())],
+    };
+
+    let mut buf = Vec::<u8>::new();
+    let mut writer = BufWriter::new(&mut buf);
+    value.write(&mut writer, false).expect("I CAN WRITE");
```

Comment:
printing things for human to check usually is not the best automated testing approach ;-)